sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -392,6 +392,8 @@ object DataType {
               equalsIgnoreCaseAndNullability(l.dataType, r.dataType)
           }
 
+      case (TimestampNTZType, TimestampType) => true
```

Comment:
This is definitely not OK. We can't change the semantic of `equalsIgnoreCaseAndNullability`